traitlets/traitlets.py

```diff
@@ -776,38 +800,33 @@ def _notify_trait(self, name, old_value, new_value):
         callables.extend(self._trait_notifiers.get('anytrait', []))
 
         # Now static ones
-        try:
-            cb = getattr(self, '_%s_changed' % name)
-        except:
-            pass
-        else:
+        if hasattr(self, '_%s_changed' % name):
```

Comment:
This should check if it's an ObserveHandler before showing a deprecation warning. Otherwise

```python
@observe('a')
def _a_changed(self, change)
```

will show an inappropriate deprecation error